test/unit-test/FreeRTOS_DNS/FreeRTOS_DNS_stubs.c

```diff
@@ -0,0 +1,109 @@
+/* Include Unity header */
+#include <unity.h>
+
+/* Include standard libraries */
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include "FreeRTOS.h"
+#include "task.h"
+#include "list.h"
+
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+
+const BaseType_t xBufferAllocFixedSize = pdFALSE;
+uint16_t usPacketIdentifier = 0U;
+volatile BaseType_t xInsideInterrupt = pdFALSE;
```

Comment:
Are these unit tests running on a server or an MCU?  Will there be real interrupts?

Reply:
No, there will not be real interrupts. But this variable is defined in one of the modules which are not compiled (but mocked). Thus, to allow the test to compile, this variable needs to be defined in stubs.